src/infrastructure/parser.rs

```
@@ -96,12 +142,15 @@ impl SplitStringParser {
                 let atom = match (int_parse_result, float_parse_result) {
                     (Ok(value), _) => Integer(value),
                     (_, Ok(value)) => Float(value),
-                    (_, _) => return Err("Parse error. This should never happen".to_string())
+                    (_, _) => return Err("Parse error. This should never happen".to_string()),
```

Can we have a more meaningful error message here? What has to happen to get here?

- [ ] to do

---

i believe if we try to parse something like 1231231231231231231231231 that would do the trick. if the number we supplied doesnt fit the i64 or f64 it should throw this error.